### ISSUE TITLE
Separate Broker Runtime Context from Broker Startup/Shutdown Context

### ISSUE THREAD

**Issue by pihme** [camunda-cloud/zeebe#7540]
**Description**
Depends on #7539 

Outcome of #7539 is, among others, the context interface that is needed during startup/shutdown.

* Based on this, define a smaller interface that contains only getters. This interface will be the broker runtime context. (see also `PartitionBootstrapContext` vs. `PartitionContext` for an idea of what the goal is)
* Use this interface for anyone who needs access to application-scoped components; limit the usage of `BrokerBootstrapContext` to Broker and the startup steps
* Look for overlap between `PartitionBootstrapContext` and `BrokerRuntimeContext`. It might be easier/better to understand if `PartitionBootstrapContext` would have a getter for the `BrokerRuntimeContext` instead of several getters for mesaging service, snapshot store, exporter repository, broker config, etc.; 
  * if this seems like a fruitful simplification, refactor the partition interface and call chain accordingly
  * If not, think of other ways how `PartitionBootstrapContext` could be simplified, by putting all the application-wide dependencies into a common context object; create new issue if needed.

However, it can also be that we don't need a broker runtime context, because there is nothing that is triggered, after all startup steps are completed. In this case, this issue is void.

**Comment by pihme** [comment]
Issues was implemented a little differently than initially intended. There is no use for the `BrokerContext` after the startup, because after startup all components have already been initialized. So all dependencies have been resolved beforehand.

The one place where the `BrokerContext` is used in tests which want to look up components inside the broker.
